tools/runner

```diff
@@ -115,6 +116,8 @@ try:
             test_params.setdefault('top_module', '')
             test_params.setdefault('timeout', "10")
             test_params.setdefault('type', 'parsing')
+            test_params.setdefault('should_fail', "0")
+            test_params.setdefault('should_fail_because', "")
```

Comment:
We should check here if `should_fail_because` is not empty and set default `should_fail` accordingly ie. when someone provides only `should_fail_because` in a test description we will still set `should_fail` to `0` and get a warning in the check further down

Reply:
here i set the default value if not assigned, and after the for/else, i do the check as you said in your previous [comment](for/else) [line 132](https://github.com/SymbiFlow/sv-tests/pull/687/files#diff-3ff66653bd0ebf86aca23ca3ef75b58bR132)
so if those are not set they will be set to the default case should_fail:0 and should_fail_because:""
if one of them is set the check after the for/else release appropriate warning and set shoud_fail:1 if necessary.

Reply:
I don't think that we should issue a warning if `should_fail_because` was set in the file and a default `should_fail` was used because in the long run this is how tests will be tagged ie. set `should_fail_because` for test that are expected to fail and don't bother with manually adding `should_fail`

Reply:
yes right, i misread your 1st comment, then reread it and deleted mine, 
I did the check at first but removed it as i thought it was redundant, anyway i updated and pushed the last modifications